### PR TITLE
Feat: Introduce the table diff API endpoint

### PR DESCRIPTION
**Pull request by vchan**

### CODE REVIEW

File: tests/web/test_main.py [hunk 1]
```diff
@@ -516,3 +516,24 @@ def test_get_lineage(web_sushi_context: Context) -> None:
...
+        no_prompts=True,
+        auto_apply=True,
+        skip_tests=True,
+    )
+    response = client.get(
+        "/api/table_diff",
+        params={
+            "source": "prod",
+            "target": "dev",
+            "on": "s.id = t.id",
```
Comment (tobymao) [inline tests/web/test_main.py:533]: while you're add it, can you add the automatic usage of "grain" to the table diff? so there's now a concept of grain which will populate the on field auto, so you won't need to set this manually

Comment (vchan) [reply tests/web/test_main.py:533]: Yea, I will add it.

File: web/server/models.py [hunk 2]
```diff
@@ -259,3 +260,37 @@ class ApiExceptionPayload(BaseModel):
...
+    modified: t.Dict[str, str]
+
+    @validator("source_schema", "target_schema", "added", "removed", "modified", pre=True)
+    def validate_schema(
+        cls, v: t.Union[t.Dict[str, exp.DataType], t.List[t.Tuple[str, exp.DataType]]]
+    ) -> t.Dict[str, str]:
+        if isinstance(v, dict):
+            return {k: str(v) for k, v in v.items()}
+        if isinstance(v, list):
+            return {k: str(v) for k, v in v}
```
Comment (tobymao) [inline web/server/models.py:281]: what if it doesn't go through this block?

**Review by tobymao — APPROVED** [review]

File: web/server/models.py (hunk 2 above)
Comment (vchan) [reply web/server/models.py:282]: Good catch.

**Pull request merged**